vignettes/upload.Rmd

```diff
@@ -0,0 +1,108 @@
+---
+title: "File Upload"
+author: "KimYen Ladia"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{File Upload}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+## File Upload
+
+Files in Synapse are versionable. Please see [Files and Versioning](https://docs.synapse.org/articles/files_and_versioning.html) for more information about how versions in Files works.
+
+Preliminaries:
+```{r collapse=TRUE}
+library(synapser)
+synLogin()
+# Create a new project
+# use hex_digits to generate random string
+hex_digits <- c(as.character(0:9), letters[1:6])
+projectName <- sprintf("My unique project %s", paste0(sample(hex_digits, 32, replace = TRUE), collapse = ""))
+project <- Project(projectName)
+project <- synStore(project)
+
+# Create a file
+file_path <- tempfile("testUpload.txt")
+connection <- file(file_path)
+writeChar("this is the content of the file", connection, eos = NULL)
+close(connection)
+file <- File(path = file_path, parent = project)
+file <- synStore(file)
+file_id <- file$properties$id
+```
+
+### Uploading a New Version
+
+Uploading a new version follows the same steps as uploading a file for the first time - use the same file name and store it in the same location (e.g., the same parentId). It is recommended to add a comment to the new version in order to easily track differences at a glance. The example file raw_data.txt will now have a version of 2 and a comment describing the change.
+
+*Explicit example:*
+```{r collapse=TRUE}
+# fetch the file in Synapse
+file_to_update <- synGet(file_id, downloadFile = FALSE)
+
+# create the second version as a separated file
+file_path <- tempfile("testUpload.txt")
+connection <- file(file_path)
+writeChar("this is version 2", connection, eos = NULL)
+close(connection)
+
+# save the local path to the new version of the file
+file_to_update$path <- file_path
+
+# add a version comment
+file_to_update$versionComment <- 'change the file content'
+
+# store the new file
+updated_file <- synStore(file_to_update)
+```
+
+*Implicit example:*
+```{r collapse=TRUE}
+# To create a new version of that file, make sure you store it with the exact same name
+new_file <- synStore(File(file_path,  parentId = project$properties$id))
+```
+
+### Updating Annotations or Provenance without Changing Versions
+
+Any change to a File will automatically update its version. If this isn’t the desired behavior, such as minor changes to the metadata, you can set forceVersion=FALSE with the Python client. For command line, the commands set-annotations and set-provenance will update the metadata without creating a new version. Adding/updating annotations and provenance in the web client will also not cause a version change.
```

Comment:
remove: "For command line, the commands set-annotations and set-provenance will update the metadata without creating a new version."